.github/workflows/release.yml

```diff
@@ -0,0 +1,360 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - .*
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build:
+    name: ${{ matrix.platform.name }} ${{ matrix.config.name }}
+    runs-on: ${{ matrix.platform.os }}
+
+    env:
+      CMAKE_CXX_COMPILER_LAUNCHER: ccache # Use ccache to cache C++ compiler output
+
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+        - { name: Windows VS2022 x86, os: windows-2025 }
+        - { name: Windows VS2022 x64, os: windows-2025 }
+        - { name: Windows MinGW x86,  os: windows-2025, flags: -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ }
+        - { name: Windows MinGW x64,  os: windows-2025, flags: -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ }
+        - { name: Linux GCC,          os: ubuntu-24.04 }
+        - { name: macOS x64,          os: macos-13 }
+        - { name: macOS arm64,        os: macos-15 }
+        config:
+        - { name: Shared, flags: -DBUILD_SHARED_LIBS=ON }
+        - { name: Static, flags: -DBUILD_SHARED_LIBS=OFF }
+        type:
+        - { name: Release }
+        - { name: Debug, flags: -DCMAKE_BUILD_TYPE=Debug }
```

Comment:
Debug libs help with MSVC since you cannot mix Release and Debug but for all other platforms it may be fine otherwise only ship Release since Clang and GCC are happy to link Release libs to a Debug application.

Reply:
Removed debug for macOS and Linux

I think it also doesn't work for MinGW, due to the runtime usage